### ISSUE TITLE
Error saving configuration profile

### ISSUE THREAD

**Issue by neives** [erikberglund/ProfileCreator#48]
Can not save my profile as ConfigurationProfile.
I'm getting an error: "The document "profilename" could not be saved as "filename"

**Comment by erikberglund** [comment]
Thanks for the info, this could be anything from a poor error message to a bug. If you don't have anything secret in the profile, could you right click the profile in the main window and select **"Show "\<ProfileName>" in Finder"** and then post that file here so I can test the same configuration on my end.

**Comment by neives** [comment]
`<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>EditorSettings</key>
	<dict>
		<key>DisableOptionalKeys</key>
		<true/>
		<key>DistributionMethod</key>
		<string>Any</string>
		<key>PayloadContentStyle</key>
		<string>Profile</string>
		<key>PlatformIOS</key>
		<false/>
		<key>PlatformMacOS</key>
		<true/>
		<key>PlatformTvOS</key>
		<false/>
		<key>ScopeSystem</key>
		<true/>
		<key>ScopeSystemManaged</key>
		<true/>
		<key>ScopeUser</key>
		<true/>
		<key>ScopeUserManaged</key>
		<true/>
		<key>ShowDisabledKeys</key>
		<true/>
		<key>ShowHiddenKeys</key>
		<false/>
		<key>ShowSupervisedKeys</key>
		<true/>
		<key>ShowUserApprovedKeys</key>
		<true/>
		<key>SignProfile</key>
		<false/>
	</dict>
	<key>Identifier</key>
	<string>08173B42-5D9B-4FA8-957C-B2B5401411A8</string>
	<key>PayloadSettings</key>
	<dict>
		<key>1</key>
		<dict>
			<key>com.apple.general</key>
			<array>
				<dict>
					<key>PayloadDescription</key>
					<string>Force password with 4char</string>
					<key>PayloadDisplayName</key>
					<string>macOS - PasswordPolicy</string>
					<key>PayloadIdentifier</key>
					<string>com.osit-group.08173B42-5D9B-4FA8-957C-B2B5401411A8</string>
					<key>PayloadOrganization</key>
					<string>Osit</string>
					<key>PayloadUUID</key>
					<string>08173B42-5D9B-4FA8-957C-B2B5401411A8</string>
					<key>PayloadVersion</key>
					<integer>1</integer>
				</dict>
			</array>
			<key>com.apple.mobiledevice.passwordpolicy</key>
			<array>
				<dict>
					<key>PayloadUUID</key>
					<string>720733DB-36E5-4E89-BA18-6B5897DCB7F7</string>
					<key>PayloadVersion</key>
					<integer>1</integer>
					<key>minLength</key>
					<string>4</string>
					<key>requireAlphanumeric</key>
					<true/>
				</dict>
			</array>
		</dict>
	</dict>
	<key>Title</key>
	<string>macOS - PasswordPolicy</string>
	<key>ViewSettings</key>
	<dict>
		<key>1</key>
		<dict>
			<key>com.apple.general</key>
			<dict>
				<key>Enabled</key>
				<true/>
				<key>Settings</key>
				<array>
					<dict>
						<key>PayloadDescription</key>
						<dict>
							<key>Enabled</key>
							<true/>
						</dict>
						<key>PayloadOrganization</key>
						<dict>
							<key>Enabled</key>
							<true/>
						</dict>
					</dict>
				</array>
			</dict>
			<key>com.apple.mobiledevice.passwordpolicy</key>
			<dict>
				<key>Settings</key>
				<array>
					<dict>
						<key>allowSimple</key>
						<dict>
							<key>Enabled</key>
							<true/>
						</dict>
						<key>forcePIN</key>
						<dict>
							<key>Enabled</key>
							<true/>
						</dict>
						<key>minLength</key>
						<dict>
							<key>Enabled</key>
							<true/>
						</dict>
						<key>requireAlphanumeric</key>
						<dict>
							<key>Enabled</key>
							<true/>
						</dict>
					</dict>
				</array>
			</dict>
		</dict>
	</dict>
</dict>
</plist>
`

**Comment by neives** [comment]
`<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>EditorSettings</key>
	<dict>
		<key>DisableOptionalKeys</key>
		<true/>
		<key>DistributionMethod</key>
		<string>Any</string>
		<key>PayloadContentStyle</key>
		<string>Profile</string>
		<key>PlatformIOS</key>
		<false/>
		<key>PlatformMacOS</key>
		<true/>
		<key>PlatformTvOS</key>
		<false/>
		<key>ScopeSystem</key>
		<true/>
		<key>ScopeSystemManaged</key>
		<true/>
		<key>ScopeUser</key>
		<true/>
		<key>ScopeUserManaged</key>
		<true/>
		<key>ShowDisabledKeys</key>
		<true/>
		<key>ShowHiddenKeys</key>
		<false/>
		<key>ShowSupervisedKeys</key>
		<true/>
		<key>ShowUserApprovedKeys</key>
		<true/>
		<key>SignProfile</key>
		<false/>
	</dict>
	<key>Identifier</key>
	<string>08173B42-5D9B-4FA8-957C-B2B5401411A8</string>
	<key>PayloadSettings</key>
	<dict>
		<key>1</key>
		<dict>
			<key>com.apple.general</key>
			<array>
				<dict>
					<key>PayloadDescription</key>
					<string>Force password with 4char</string>
					<key>PayloadDisplayName</key>
					<string>macOS - PasswordPolicy</string>
					<key>PayloadIdentifier</key>
					<string>com.osit-group.08173B42-5D9B-4FA8-957C-B2B5401411A8</string>
					<key>PayloadOrganization</key>
					<string>Osit</string>
					<key>PayloadUUID</key>
					<string>08173B42-5D9B-4FA8-957C-B2B5401411A8</string>
					<key>PayloadVersion</key>
					<integer>1</integer>
				</dict>
			</array>
			<key>com.apple.mobiledevice.passwordpolicy</key>
			<array>
				<dict>
					<key>PayloadUUID</key>
					<string>720733DB-36E5-4E89-BA18-6B5897DCB7F7</string>
					<key>PayloadVersion</key>
					<integer>1</integer>
					<key>minLength</key>
					<string>4</string>
					<key>requireAlphanumeric</key>
					<true/>
				</dict>
			</array>
		</dict>
	</dict>
	<key>Title</key>
	<string>macOS - PasswordPolicy</string>
	<key>ViewSettings</key>
	<dict>
		<key>1</key>
		<dict>
			<key>com.apple.general</key>
			<dict>
				<key>Enabled</key>
				<true/>
				<key>Settings</key>
				<array>
					<dict>
						<key>PayloadDescription</key>
						<dict>
							<key>Enabled</key>
							<true/>
						</dict>
						<key>PayloadOrganization</key>
						<dict>
							<key>Enabled</key>
							<true/>
						</dict>
					</dict>
				</array>
			</dict>
			<key>com.apple.mobiledevice.passwordpolicy</key>
			<dict>
				<key>Settings</key>
				<array>
					<dict>
						<key>allowSimple</key>
						<dict>
							<key>Enabled</key>
							<true/>
						</dict>
						<key>forcePIN</key>
						<dict>
							<key>Enabled</key>
							<true/>
						</dict>
						<key>minLength</key>
						<dict>
							<key>Enabled</key>
							<true/>
						</dict>
						<key>requireAlphanumeric</key>
						<dict>
							<key>Enabled</key>
							<true/>
						</dict>
					</dict>
				</array>
			</dict>
		</dict>
	</dict>
</dict>
</plist>
`

**Comment by neives** [comment]
Sure.

Here you go: https://pastebin.com/L9EdLxAt

**Comment by erikberglund** [comment]
Thank you. There were two things: 

1. I found a bug in the popUpButton implementation used by the key *minLength* that I will fix.

2. In the config you sent me, no payload was included in the profile. And that is a thing that's not communicated or shown how to be done anywhere. I don't know if that was the case for you, but the config I loaded looked like this. (see screenshot) And in the area below the "General" settings where I have marked, the payloads that will be exported has to be added to. Either by pressing the add button that is shown in the top right corner when a payload (other than General) is selected or by drag/drop.

![screen shot 2018-07-09 at 13 01 42](https://user-images.githubusercontent.com/1296345/42447070-d8e7a9d2-8378-11e8-8e4e-95dcbbcaeaba.png)

This is something I know is not said anywhere and a user just has to figure it out. So I have that on my list to somehow make more obvious.

But, even if I added the passcode policy payload, the bug still prevented me from exporting.

**Comment by neives** [comment]
Sorry, I missed the payload on the profile I've sent you, but the other times I run he was there.
I'm testing with other configs/payloads/profiles, and I can't save none of them.

Ok, I'm also having key minLength issue.

**Comment by DarthAndersen** [comment]
I had the same problem and I think maybe it's down to a simple usability issue. If you choose "Save as" (thinking it'll let you export the profile) with the profile document window active and set the format to "Configuration Profile" you'll get the incomprehensible "could not save" message.
I'm guessing @neives wanted to output a .mobileconfig file (like I did) and the way is to do it is to right click and choose "Export" in the listing on the Main Window

**Comment by erikberglund** [comment]
Yeah, I talked to another person on the MacAdmins slack about this issue and found the Save As confusion. I have removed the Save As menu item in the next update to avoid this. And from how the application is designed it is not needed. I will add export options instead to the menu and later to each profile editor window to make it easier to find and use. Don't know if it makes it in the next update but definitely the one after.
